File: pfm-ui/src/main/java/com/tink/pfmui/view/BaseLineHelper.java
package com.tink.pfmui.view;

class BaseLineHelper {

	private final float materialStep;

	BaseLineHelper(float materialStep) {
		this.materialStep = materialStep;
	}

	int putHeightOnGrid(int height) {
		float offsetFromGrid = height % materialStep;
		if (offsetFromGrid > 0) {
			return (int) (materialStep - Math.floor(offsetFromGrid));
		}
		return 0;
	}


	int getCompoundTopPaddingToPutBaselineOnGrid(int baseLine) {
		float offsetFromGrid = baseLine % materialStep;
		if (offsetFromGrid > 0) {
			return (int) (materialStep - Math.ceil(offsetFromGrid));
		}
		return 0;
	}
}

File: moneymanager-ui/src/main/java/com/tink/moneymanagerui/view/BaseLineHelper.kt
package com.tink.moneymanagerui.view

import kotlin.math.ceil
import kotlin.math.floor

internal class BaseLineHelper(private val materialStep: Float) {
    fun putHeightOnGrid(height: Int): Int {
        val offsetFromGrid = height % materialStep
        return if (offsetFromGrid > 0) {
            (materialStep - floor(offsetFromGrid.toDouble())).toInt()
        } else 0
    }

    fun getCompoundTopPaddingToPutBaselineOnGrid(baseLine: Int): Int {
        val offsetFromGrid = baseLine % materialStep
        return if (offsetFromGrid > 0) {
            (materialStep - ceil(offsetFromGrid.toDouble())).toInt()
        } else 0
    }
}